Fixes navigation bug

AlexSCorey opened the pull request:
##### SUMMARY
This PR addresses a bug that @dsesami found in the Project JobTemplate list item navigation.  The bug was that when you click on the list item it would not navigate to the correct page and would append the pathname to the end of the url. This fixes that, and adds tests to catch that same issue in the future.  https://github.com/ansible/awx/issues/5777 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
 - UI

##### AWX VERSION
```
9.2.1
```


##### ADDITIONAL INFORMATION
![project navigation](https://user-images.githubusercontent.com/39280967/74975482-1bfed580-53f5-11ea-9293-8baa2e36f912.gif)

AlexSCorey committed:
Fixes navigation bug

jakemcdermott commented on awx/ui_next/src/screens/Template/TemplateList/TemplateListItem.jsx:
```diff
@@ -31,6 +31,12 @@ function TemplateListItem({ i18n, template, isSelected, onSelect, detailUrl }) {
+  const location = useLocation();
+
+  if (location.pathname.startsWith('/projects')) {
+    detailUrl = `/templates/job_template/${template.id}/details`;
```
I don't think a lower-level list item component should be smart enough to introspect what view it is on.

Instead of inspecting the URL, could you instead pass a different `detailUrl` from the parent project component?

jakemcdermott replied:
edit: To to this, we'll first need to do: https://github.com/ansible/awx/pull/6015#issuecomment-589298424

Let's make that a part of this PR.

AlexSCorey replied:
@jakemcdermott We do have similar behavior in Inventory Host where we borrow the Host list, and form and use it within the Inventory context.  Do you think that should be redone as well?

jakemcdermott replied:
No. Let's fix that later and not in this PR.

jakemcdermott commented:
https://github.com/ansible/awx/blob/devel/awx/ui_next/src/screens/Project/ProjectJobTemplates/ProjectJobTemplates.jsx#L6

`screens` shouldn't import from each other.

I'd like for us to not re-use the `../../Template/TemplateList/TemplateList` in the way that we're currently doing. My recommendation is to 

1. duplicate the contents of `TemplateList` in `ProjectJobTemplates` 

or 

2. move `TemplateList` to a shared component in `src/components`.

I'd prefer option 1 right now unless there's other places we'll be needing a related templates list in the near future.

keithjgrant commented on awx/ui_next/src/screens/Template/TemplateList/TemplateListItem.test.jsx:
```diff
@@ -161,4 +162,88 @@ describe('<TemplateListItem />', () => {
+      initialEntries: ['/templates'],
+    });
+    const wrapper = mountWithContexts(
+      <Route
```
does this need the <Route> wrapping it for history changes to work? I don't think we've had to do that for similar tests in other components

AlexSCorey replied:
That is how I have done it elsewhere in our project, but further exploration says no, we don't need Route.

AlexSCorey committed:
WIP

AlexSCorey committed:
Fixes naviation bug by create ProjectTemplateList and ProjectTemplateListItem

Adds tests for those new files and removes uncessary test from TemplateListItem

AlexSCorey committed:
prettier

dsesami reviewed (approved):
Works great, links are functional. Thanks for this
